pkg/util/chunking.go

```diff
@@ -1,7 +1,13 @@
 package util
 
+import "fmt"
+
 // ForEachChunk executes the given handler for each chunk of items in the slice.
 func ForEachChunk[T any](data []T, chunkSize uint64, handler func(items []T)) {
+	if chunkSize <= 0 {
+		panic(fmt.Sprintf("invalid chunk size: %d", chunkSize))
```

Comment:
I wonder if we need to panic. It's clearly an error, but can we recover from it by ignoring that value and setting it to 1, and logging the error?

I'm just growing wary of the panics.

Reply:
Sure